packages/glimmer-test-helpers/lib/environment.ts

```diff
@@ -441,14 +452,18 @@ class EmberishCurlyComponentManager implements ComponentManager<EmberishCurlyCom
     return component;
   }
 
-  layoutFor(definition: EmberishCurlyComponentDefinition, component: EmberishCurlyComponent, env: Environment): CompiledBlock {
+  layoutFor(definition: EmberishCurlyComponentDefinition, component: EmberishCurlyComponent, env: TestEnvironment): CompiledBlock {
     let layoutString = definition.layoutString;
 
     if (!layoutString && layoutString !== '') {
       layoutString = component['layout'];
     }
```

Comment:
This is the late-bound layout case – so we either cannot cache it (just do a early return here), or need to change the cache key to account for that.

If the goal of this is to not regress demos and benchmarks, it's probably fine to not cache late-bound layouts – I don't think that feature is used outside of tests.